Reduce diff of prater and mainnet package

nflaig opened the pull request:
Reduce diff of prater and mainnet package to increase testing confidence and reduce maintenance efforts.

See commits for individual changes done

Related PR in mainnet package https://github.com/dappnode/DAppNodePackage-Lodestar/pull/44

nflaig committed:
Fix links in dappnode package file

nflaig committed:
Fix typo in readme

nflaig committed:
Set log level from DEBUG_LEVEL env

nflaig committed:
Enable doppelganger protection

nflaig committed:
Only check if mev-boost is reachable in validator entrypoint script

nflaig commented on beacon-chain/entrypoint.sh:
```diff
@@ -22,16 +22,11 @@ case $_DAPPNODE_GLOBAL_EXECUTION_CLIENT_PRATER in
-    echo "MEVBOOST is enabled"
+    echo "MEV-Boost is enabled"
     MEVBOOST_URL="http://mev-boost.mev-boost-goerli.dappnode:18550"
-    if curl --retry 5 --retry-delay 5 --retry-all-errors "${MEVBOOST_URL}"; then
```
all other packages only check if mev-boost is reachable in validator entrypoint script, I would assume a user would get two notifications if we have this check also in the beacon-chain entrypoint script

alexpeterson91 replied:
good catch however the outcome would be the same, it would only give 2 notifications when MEV-Boost is enabled but not connected to either service of the CL.  none if it is working. 
Thinking further though, all CLs need MEV-Boost params entered in both services (sans Nimbus the monoservice), and need both services to connect to MEV-Boost in order to work properly, so i think we should keep the check here and add checks to the beacon chains of lighthouse, Prysm, and Teku, otherwise it may not connect on the beaconchain service but the user would get no notification if this failure to connect.   @pablomendezroyo What do you think?

nflaig replied:
would be in favor of just removing it in this PR and just add it again later in a separate PR for both prater and mainnet package once we are clear how to ideally handle this.

The main goal right now is ensure stability of the Lodestar package for the release we want to ideally do this week. By decreasing the difference between Lodestar package and other CL packages we reduce the risk of running into unexpected issues. Same goes for prater and mainnet package, the difference should be minimal.

nflaig replied:
@alexpeterson91 I moved the mev-boost health check to the beacon node, this makes much more sense as the validator doesn't even communicate with the builder directly, only the beacon node does.

nflaig committed:
Remove HTTP_ENGINE env from docker-compose file

nflaig committed:
Remove CORSDOMAIN env from docker-compose file

nflaig committed:
Move mev-boost health check to beacon node entrypoint script

Only the beacon node needs to communicate with the builder (mev-boost)
and in case mev-boost is not running we only want to send one notification
to the user.

See https://github.com/dappnode/DAppNodePackage-Lodestar-Prater/pull/52#discussion_r1133072650

nflaig committed:
Set file log level to debug (default value)

nflaig committed:
Remove MEVBOOST_URL variable from validator entrypoint

dsimog01 commented on dappnode_package.json:
```diff
@@ -34,7 +34,7 @@
     "driver": "ethereum-beacon-chain",
     "serviceName": "beacon-chain",
-    "port": 3500
+    "portNumber": 3500
```
Good catch!

dsimog01 reviewed (approved):
LGTM
